web/packages/teleterm/src/ui/DocumentTerminal/useDocumentTerminal.ts

```diff
@@ -389,6 +391,40 @@ function createCmd(
     };
   }
 
+  if (doc.kind === 'doc.gateway_kube') {
+    const gateway = clustersService.findGatewayByConnectionParams(
+      doc.targetUri,
+      ''
+    );
+    if (!gateway) {
+      throw new Error(`No gateway found for ${doc.targetUri}`);
+    }
+
+    const env = tshdGateway.getCliCommandEnv(gateway.gatewayCliCommand);
+
+    if ('KUBECONFIG' in env === false) {
+      // This shouldn't happen as 'KUBECONFIG' is the sole purpose of the CLI
+      // command for a kube gateway.
+      throw new Error(
+        `No KUBECONFIG provided for gateway ${gateway.targetUri}`
+      );
+    }
+    const helpMsg =
+      "Started local proxy for Kubernetes cluster '" +
+      gateway.targetName +
+      "' and environment variable 'KUBECONFIG' is updated to the kubeconfig provided by the local proxy.\r\n\r\n" +
+      "Try 'kubectl version' to test the connection.\r\n\r\n";
+
+    return {
+      ...doc,
+      kind: 'pty.shell',
```

Comment:
Do you remember if we talked about persisting cwd and what was the outcome of the discussion?

Reply:
No, I don't recall we have talked about cwd. is it updated in `refreshTitle`?

Reply:
Ok, I found it. https://github.com/gravitational/teleport/pull/27287#discussion_r1228154159

It looks like everything is according to what we discussed back then.

Yeah, it is updated in `refreshTitle`, but properly handling this requires also storing it on a document and a couple of other things, we can ignore this for now.